### PR TITLE
Add logging eth_getLogs errors #4909

### LINKED ISSUE #4909: Add logging `eth_getLogs` errors

### PR DESCRIPTION
**Pull request by oa-s**
Closes #4909

### CODE REVIEW

File: AlphaWallet/Tokens/Helpers/CallSmartContractFunction.swift [hunk 1]
```diff
@@ -141,4 +141,8 @@ func getEventLogs(
     }
 
     return contractInstance.getIndexedEventsPromise(eventName: eventName, filter: filter)
+        .recover { error -> Promise<[EventParserResultProtocol]> in
+            debugLog("[eth_getLogs] failure for server: \(server) with error: \(error)")
```
Comment (hboon) [inline AlphaWallet/Tokens/Helpers/CallSmartContractFunction.swift:145]: @oa-s Have you seen why/when it happens?

Comment (oa-s) [reply AlphaWallet/Tokens/Helpers/CallSmartContractFunction.swift:145]: in my case i got this when received an error like `PromiseKit:cauterized-error: nodeError("exceed maximum block range: 5000")`, so this pr, help identify server where an error occured

Comment (hboon) [reply AlphaWallet/Tokens/Helpers/CallSmartContractFunction.swift:145]: Ah, like the one for mumbai? Great!

Comment (oa-s) [reply AlphaWallet/Tokens/Helpers/CallSmartContractFunction.swift:145]: yes right, i also cached similar with some other network.

**Pull request merged**